sros2/test/sros2/test_cli.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 Apex.AI, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import tempfile
+
+from lxml import etree
+import pytest
+
+
+def test_cli_without_network():
+    # Test that the CLI tool still works even when libxml2 can't look up external resources
+    # because its been set to use a proxy that won't resolve.
+    # This test regressions of https://github.com/ros2/sros2/issues/157
+
+    env = os.environ.copy()
+    env['HTTP_PROXY'] = 'http://example.com'
+    tmp_dir = tempfile.mkdtemp(prefix='sros2_test')
+
+    subprocess.run(
+        args=['ros2', 'security', 'create_keystore', 'demo_keys'],
+        check=True,
+        cwd=tmp_dir,
+        env=env
```

Comment:
If switching to calling the command via the api, perhaps you could use the same env_backup pattern you have bellow to set/restore the env before/after setup and teardown.